crates/neon/src/context/mod.rs

```diff
@@ -593,6 +618,41 @@ impl<'a> ModuleContext<'a> {
         Ok(())
     }
 
+    #[cfg(feature = "async_local")]
+    pub fn export_function_async<'b, F, V, Fut>(&mut self, key: &str, f: F) -> NeonResult<()>
+    where
+        Fut: Future<Output = JsResult<'b, V>>,
+        F: Fn(AsyncFunctionContext) -> Fut + 'static + Copy,
+        V: Value,
+    {
+        use crate::handle::StaticHandle;
+
+        let wrapper = JsFunction::new(self, move |mut cx| {
+            let mut args = vec![];
+
+            while let Some(arg) = cx.argument_opt(args.len()) {
+                let arg = arg.as_value(&mut cx);
+                let arg = StaticHandle::new(&mut cx, arg)?;
+                args.push(arg);
+            }
+
+            let (deferred, promise) = cx.promise();
+            cx.spawn_local(move |mut cx| async move {
+                let acx = AsyncFunctionContext {
+                    env: cx.env(),
+                    arguments: args,
+                };
+                deferred.resolve(&mut cx, f(acx).await.unwrap());
+                ()
+            });
+
+            Ok(promise)
+        })?;
+
+        self.exports.clone().set(self, key, wrapper)?;
+        Ok(())
+    }
```

Comment:
This essentially creates:
```typescript
function(...args: Array<Root>): Promise<R>
```
I included this as a means to test the futures executor out so I can exclude it from the first PR. It is missing methods and other stuff so it's certainly not production ready.